src/main/java/com/gmail/inverseconduit/utils/SourceCodeParser.java

```diff
@@ -0,0 +1,110 @@
+package com.gmail.inverseconduit.utils;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Reads a websites source code and returns a List of matches of a search string
+ * @author Michel v. Varendorff
+ *
+ */
+public class SourceCodeParser {
+
+	String website;
+	String searchRegex;
+
+	public SourceCodeParser(String website, String searchRegex){
+		this.website = website;
+		this.searchRegex = searchRegex;
+	}
+	
+	public SourceCodeParser() {	}
+	
+	/**
+	 * Sets the website to search through
+	 * @param website URL of the website
+	 */
+	public void setWebsite(String website) {
+		this.website = website;
+	}
+
+	/**
+	 * Sets a regex that will be looked for in the websites DOM
+	 * @param searchRegex Regex to search matches for
+	 */
+	public void setSearch(String searchRegex) {
+		this.searchRegex = searchRegex;
+	}
+	
+	/**
+	 * Loads the websites code and returns the matches
+	 * @return List of all matches of the search
+	 */
+	public List<String> parse(){
+		String websiteCode = readWebsiteCode();
+		return getAllMatches(websiteCode);
+	}
+
+	/**
+	 * Reads the code of the website into one single String
+	 * @return Websites DOM as String
+	 */
+	private String readWebsiteCode(){
+		InputStream is = null;
+		BufferedReader br = null;
+		
+		String htmlCode = "";
+		
+		try {
+			String line;
+			
+			URL url = new URL(website);
+			is = url.openStream();
+			br = new BufferedReader(new InputStreamReader(is));
+			
+			while ( (line = br.readLine()) != null) {
+				htmlCode += line;
+			}
+		} catch (MalformedURLException mue){
+			mue.printStackTrace();
+		} catch (IOException ioe) {
+			ioe.printStackTrace();
+		} finally {
+			try {
+				if (is != null) is.close();
+				if (br != null) br.close();
+			} catch (IOException ioe) {
+				
+			}
+		}
+		return htmlCode;
+	}
+	
+	/**
+	 * Finds all matches of the search regex in the input String
+	 * @param htmlCode Code to look for matches
+	 * @return List of all Matches in the String
+	 */
+	private List<String> getAllMatches(String htmlCode) {

```

Comment:
I know that answer, but in other answers, it says that regex is fine to parse a limited and known set of html. In addition to that at least the fetching of the room owners has nothing to do directly with html, but looking for a pattern that shows the id's of the roomowner. This pattern completely ignores the tags. And I know that the way of getting the room name is not the clean (lets be honest, it is horrible...). But I could not find a way of finding stuff in the DOM of a website.
At first I thought of passing message.getRoomName() to the createRoom method, but this would then ignore the rooms that the bot joins during startup. 
Is there something to find a certain string pattern in HTML except for RegExes? Or should I just leave out the name out of the files?